### PR TITLE
server(cdc): adjust error handling and logs

### LINKED ISSUE #6778: adjust server error handling.

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #6778 

#### Tests 

 - No code

### Release note 

```release-note
`None`
```

### CODE REVIEW

File: cdc/capture/capture.go [hunk 1]
```diff
@@ -233,9 +229,8 @@ func (c *captureImpl) reset(ctx context.Context) error {
 		c.sorterSystem = ssystem.NewSystem(sortDir, memPercentage, c.config.Debug.DB)
 		err = c.sorterSystem.Start(ctx)
 		if err != nil {
-			return errors.Annotate(
-				cerror.WrapError(cerror.ErrNewCaptureFailed, err),
-				"create sorter system")
+			log.Error("start sorter system failed", zap.Error(err))
```
Comment (lonng) [inline cdc/capture/capture.go:232]: Some errors have been printed in the log and some of them don't output the logs. Can we unify the behavior?

**Review by asddongmen — APPROVED** [review]

**Review by sdojjy — APPROVED** [review]

**Review by overvenus — APPROVED** [review]

**Pull request merged**